### PR TITLE
add batch api v2 create batch file uri list check (WP-907)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

**Review by dimitrystd — APPROVED** [review]

File: src/Batch/BatchApiV2.php [hunk 1]
```diff
@@ -63,6 +63,21 @@ public function createBatch(
...
+            {
+                $pointer = -1;
+                foreach ($array as $key => $_) {
+                    if ($key !== $pointer) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+        }
```
Comment (PavelLoparev) [inline src/Batch/BatchApiV2.php]: I don't get it. According to php doc `An array is considered a list if its keys consist of consecutive numbers from 0 to count($array)-1.` Your pointer never gets incremented to check key against it. Your function will always return false if array doesn't start with element with key `-1`, no? Let's write tests for this function.

**Review by PavelLoparev — CHANGES_REQUESTED** [review]
Also, let's update package version as well? Like increase minor version I think

**Comment by vsolovei-smartling** [comment]
> Also, let's update package version as well? Like increase minor version I think

Let's discuss https://github.com/Smartling/api-sdk-php/commit/426cc320103fa8b2077d81e23aaf825b257d152f. Was it for a reason? Should the version return?

File: src/Batch/BatchApiV2.php (hunk 1 above)
Comment (vsolovei-smartling) [reply src/Batch/BatchApiV2.php]: Added, great catch, thanks. The tests were present, but I'm using php 8.2, so the polyfill code was not called

File: tests/unit/BatchApiV2Test.php [hunk 2]
```diff
@@ -80,4 +80,29 @@ public function testCancelBatchAction()
...
+
+    public function testArrayIsListPolyfill()
+    {
+        $this->assertTrue(BatchApiV2::array_is_list([]));
+        $this->assertTrue(BatchApiV2::array_is_list(['a', 'b', 'c']));
+        $this->assertTrue(BatchApiV2::array_is_list(array_unique(['a', 'b', 'c'])));
+        $this->assertTrue(BatchApiV2::array_is_list(array_unique([0 => 'a', 1 => 'b', 2 => 'c'])));
+        $this->assertTrue(BatchApiV2::array_is_list(array_values(array_unique(['a', 'b', 'b', 'c']))));
+        $this->assertFalse(BatchApiV2::array_is_list(array_unique(['a', 'b', 'b', 'c'])));
+        $this->assertFalse(BatchApiV2::array_is_list([1 => 'a']));
```
Comment (PavelLoparev) [inline tests/unit/BatchApiV2Test.php]: Does native php's `array_is_list` works the same?

**Review by PavelLoparev — APPROVED** [review]

File: tests/unit/BatchApiV2Test.php (hunk 2 above)
Comment (PavelLoparev) [reply tests/unit/BatchApiV2Test.php]: Just thought that this custom `array_is_list` is a generic code and not directly related to batch API. Maybe consider moving it out of `BatchApiV2` class?

**Comment by PavelLoparev** [comment]
> > Also, let's update package version as well? Like increase minor version I think
> 
> Let's discuss [426cc32](https://github.com/Smartling/api-sdk-php/commit/426cc320103fa8b2077d81e23aaf825b257d152f). Was it for a reason? Should the version return?

Good question. I don't remember. it might be related to the release process, not sure.

**Review by PavelLoparev — APPROVED** [review]

**Pull request merged**